jquery.bemhelpers.js

```diff
@@ -62,40 +62,50 @@
                         return false;
                     }
                 });
-                
+
                 return modVal;
-                
+
             } else return undefined;
         },
-        
+
         setMod: function(block, elem, modName, modVal) {
             if (modVal === undefined) {
                 modVal = modName;
                 modName = elem;
                 elem = undefined;
             }
-            
+
             return this.each(function() {
                 var $this = $(this),
                     classPattern = block + (elem !== undefined ? BEMsyntax.elem + elem : '') +
                         BEMsyntax.modBefore + modName;
-                
+
+                var removeTheAlreadySetValue = false;
+
                 if (modVal === false) {
                     $this.removeClass(classPattern);
                 } else if (modVal === true) {
                     $this.addClass(classPattern);
                 } else {
+                    // if the String modifer value is an empty String, then it means that user likes to remove it. So set 'removeTheAlreadySetValue' to true.
+                    if (modVal === '') removeTheAlreadySetValue = true;
+
                     $.each(getElemClasses(this), function(i, c) {
                         if (c.indexOf(classPattern) === 0
                             && c.substr(classPattern.length, BEMsyntax.modKeyVal.length) === BEMsyntax.modKeyVal) {
 
                             $this.removeClass(c);
                         }
                     });
-                    
-                    $this.addClass(classPattern + BEMsyntax.modKeyVal + modVal);
+
+                    // Check whether we should remove the modifer class or not according to 'removeTheAlreadySetValue'
+                    if (removeTheAlreadySetValue === true) $this.removeClass(classPattern + BEMsyntax.modKeyVal + modVal);
```

Comment:
I don't think you have to remove the class name here, as the previous loop that iterates over all the classnames already takes care of that (I think — did I miss smth?). This also means you can simplify the condition checks, too.